### PR TITLE
fix: weird date override behaviour

### LINKED ISSUE #12265: [CAL-2682] Weird date overrides behaviour
https://www.loom.com/share/df6fccf178d74d61b85ce100dfc4c66b?sid=9b30e077-de45-4397-bd1c-2a15f97b4d1d 

<sub>[CAL-2682](https://linear.app/calcom/issue/CAL-2682/weird-date-overrides-behaviour)</sub>

### PR DESCRIPTION
**Pull request by ibex088**
## What does this PR do?

Fixes #12265
## before
* if i toggle on the `mark unavailable all day` switch, it show 12:00 Am - 12:00 Am instead it should show `unavailable`
* when i toggle off the `mark unavailable all day` switch it prefills the time as 12am - 12am and if i click `update override` it shows Unavailable.

[Screencast from 09-11-23 10:45:34 AM IST.webm](https://github.com/calcom/cal.com/assets/32706411/cbe07267-44df-4271-8f8f-12bf8a0c00d7)

## after
* shows unavailabe if i toggle on  `mark unavailable all day` switch.
* when i toggle off the `mark unavailable all day` switch it prefills the time as 9am - 5pm and if i click `update override` it shows the  same..
 
[Screencast from 09-11-23 10:47:32 AM IST.webm](https://github.com/calcom/cal.com/assets/32706411/49c266e5-4667-4e70-94e9-c2d88e16740a)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: packages/features/schedules/components/DateOverrideInputDialog.tsx [hunk 1]
```diff
@@ -82,21 +82,22 @@ const DateOverrideForm = ({
...
-                .second(0)
-                .format()
-            ),
-            end: new Date(
-              dayjs.utc().hour(range.end.getUTCHours()).minute(range.end.getUTCMinutes()).second(0).format()
-            ),
-          }))
-        : defaultRanges,
+      range:
+        value && value[0].start.valueOf() - value[0].end.valueOf() !== 0
```
Comment (supalarry) [inline packages/features/schedules/components/DateOverrideInputDialog.tsx:86]: What do you think about replacing

```
value[0].start.valueOf() - value[0].end.valueOf() !== 0
```

with 
```
value[0].start.valueOf() !== value[0].end.valueOf()
```

I think it better reflects the logic of them being set as equal on line 131 and 132.

**Review by supalarry — DISMISSED** [review]
Tested locally - looks good! Thank you for your contribution ! ❤️

File: packages/features/schedules/components/DateOverrideInputDialog.tsx (hunk 1 above)
Comment (SomayChauhan) [reply packages/features/schedules/components/DateOverrideInputDialog.tsx:86]: sure thing

**Review by supalarry — APPROVED** [review]

**Comment by PeerRich** [comment]
not sure why checks are not passing @SomayChauhan can you check?

**Pull request merged**